service/routes.py

```diff
@@ -138,13 +138,32 @@ def create_product_in_inventory():
     product_in_inventory.deserialize(request.get_json())
     product_in_inventory.create()
     message = product_in_inventory.serialize()
-    location_url = url_for("get_product_in_inventory",
-                           product_in_inventory=product_in_inventory.product_in_inventory_id,
-                           _external=True)
+    # location_url = url_for("get_product_in_inventory", #FIXME I had to comment these three lines to get nosetests to pass for some reason, similar to like he said this wouldn't work in his video
+    #                        product_in_inventory=product_in_inventory.product_in_inventory_id,
+    #                        _external=True)
     location_url = "not implemented"
     return make_response(
         jsonify(message), status.HTTP_201_CREATED, {"Location": location_url})
 
+######################################################################
+# UPDATE AN EXISTING PRODUCT IN INVENTORY
+######################################################################
+@app.route("/inventory/<int:by_id>", methods=["PUT"])
+def update_product_in_inventory(by_id):
```

Comment:
I think the '''by_id''' input should be something like "inventory_id" or "product_in_inventory_id"

Reply:
"by_id" was something the donor code used, and I didn't use "product_in_inventory_id" since that was already defined in our model. He used "id" in his model and used "by_id" in his update code, so I just left it to not reuse something already defined in our model. I think inventory_id is unused so I could probably change it. Should I do that?

Reply:
in this case, `product_in_inventory_id` would be just the name of the argument that the function takes in. every function exists in its own scope so it wont get confused with what you call other things in other functions. if that makes sense.

though tbh thinking about it more i like inventory_id better all around (even for the data model)